### PR TITLE
feat(verify): add `--verbose` flag and fix flicker in survey prompt

### PR DESCRIPTION
**Pull request by siddhant-deepsource**

### CODE REVIEW

File: command/analyzer/verify/verify.go [hunk 1]
```diff
@@ -101,11 +107,11 @@ func (*AnalyzerVerifyOpts) Run() (err error) {
 
 	// Do not move to building the image if the configuration verification fails
 	if !configurationValid {
-		return
+		configurationValid = true
```
Comment (siddhant-deepsource) [inline command/analyzer/verify/verify.go:110]:
```suggestion
    return
```

File: utils/spinner.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ func (s *SpinnerUtils) StartSpinnerWithLabel(label, finalMessage string) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	sp := spinner.New(spinner.CharSets[11], 120*time.Millisecond, spinner.WithWriter(os.Stderr), spinner.WithFinalMSG(GetSuccessMessage(finalMessage))) // Build our new spinner
+	sp := spinner.New(spinner.CharSets[11], 100*time.Millisecond, spinner.WithWriter(os.Stdout), spinner.WithWriter(os.Stderr), spinner.WithFinalMSG(GetSuccessMessage(finalMessage))) // Build our new spinner
```
Comment (siddhant-deepsource) [inline utils/spinner.go:22]:
```suggestion
	sp := spinner.New(spinner.CharSets[11], 120*time.Millisecond, spinner.WithWriter(os.Stdout), spinner.WithWriter(os.Stderr), spinner.WithFinalMSG(GetSuccessMessage(finalMessage))) // Build our new spinner
```

File: analyzers/validator/validator.go [hunk 3]
```diff
@@ -80,6 +80,7 @@ func ValidateAnalyzerTOML(analyzerTOMLPath string) (*config.AnalyzerMetadata, *V
 
 	// Unmarshal TOML into config
 	if err = toml.Unmarshal(analyzerTOMLContent, &config); err != nil {
+		fmt.Println("Found an error", err)
```
Comment (sourya-deepsource) [inline analyzers/validator/validator.go:83]:
```suggestion
```

File: analyzers/validator/testdata/analyzer2/.deepsource/analyzer/analyzer.toml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name = "Todo comments checker"
-# path = "@deepsource/demo-python"
+# shortcode = "@deepsource/demo-python"
```
Comment (sourya-deepsource) [inline analyzers/validator/testdata/analyzer2/.deepsource/analyzer/analyzer.toml:2]:
```suggestion
# shortcode = "@deepsource/demo-python" # test invalid config
```

File: command/analyzer/verify/verify.go [hunk 5]
```diff
@@ -67,9 +76,10 @@ func (*AnalyzerVerifyOpts) Run() (err error) {
 	// Read and verify analyzer toml
 	spin.StartSpinnerWithLabel("Validating analyzer.toml...", "Verified analyzer.toml")
 	analyzerTOMLData, analyzerTOMLValidationErrors, err := validator.ValidateAnalyzerTOML(analyzerTOMLPath)
-	if analyzerTOMLValidationErrors != nil {
+	if len(analyzerTOMLValidationErrors.Errors) > 0 {
 		configurationValid = false
 		spin.StopSpinnerWithError("Failed to verify analyzer.toml\n", err)
+		fmt.Println(analyzerTOMLValidationErrors.Errors)
```
Comment (sourya-deepsource) [inline command/analyzer/verify/verify.go:82]:
```suggestion
```

File: command/analyzer/verify/verify.go [hunk 6]
```diff
@@ -106,33 +116,60 @@ func (*AnalyzerVerifyOpts) Run() (err error) {
...
+	/* Specifying the name of the image to be built
+	 * Set the default Dockerfile path as "Dockerfile" */
 	dockerFilePath, dockerFileName := build.GetDockerImageDetails(analyzerTOMLData)
+	dockerFilePath = "Dockerfile"
+
+	// Read config for the value if specified
+	if analyzerTOMLData.Build.Dockerfile != "" {
+		dockerFilePath = analyzerTOMLData.Build.Dockerfile
+	}
+	if analyzerTOMLData.Shortcode != "" {
```
Comment (sourya-deepsource) [inline command/analyzer/verify/verify.go:131]: Are we repeating code from `run` here?

Comment (siddhant-deepsource) [reply command/analyzer/verify/verify.go]: Removed

**Review by sourya-deepsource — APPROVED** [review]

File: analyzers/backend/docker/build.go [hunk 7]
```diff
@@ -109,10 +110,13 @@ func checkResponse(rd io.Reader) error {
 		if err != nil {
 			return err
 		}
-		if d.Stream == "" || d.Stream == "\n" || strings.TrimSuffix(d.Stream, "\n") == currentStream {
+		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
 			continue
 		}
 		currentStream = strings.TrimSuffix(d.Stream, "\n")
+		if showAllLogs {
```
Comment (sourya-deepsource) [inline analyzers/backend/docker/build.go:117]: Wouldn't it be better to return the reader, and get the client of this API (the `command/analyzer/verify` package) to print the output however it wants?

**Review by sourya-deepsource — APPROVED** [review]

**Pull request merged**